### PR TITLE
Add bing AutoSuggest Spec

### PR DESCRIPTION
**Pull request by gu-wei**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/cognitiveservices/data-plane/AutoSuggest/stable/v1.0/AutoSuggest.json [hunk 1]
```diff
@@ -0,0 +1,615 @@
...
+          "$ref": "#/definitions/QueryContext",
+          "readOnly": true
+        }
+      }
+    },
+    "SuggestionsSuggestionGroup": {
+      "x-ms-discriminator-value": "Suggestions/SuggestionGroup",
+      "type": "object",
+      "required": [
+        "name",
```
Comment (jianghaolu) [inline specification/cognitiveservices/data-plane/AutoSuggest/stable/v1.0/AutoSuggest.json]: "name" property definition is missing.

File: specification/cognitiveservices/data-plane/AutoSuggest/stable/v1.0/AutoSuggest.json [hunk 2]
```diff
@@ -0,0 +1,615 @@
...
+  "security": [
+    {
+      "apiKeyHeader": []
+    }
+  ],
+  "paths": {
+    "/Suggestions": {
+      "get": {
+        "summary": "The AutoSuggest API lets you send a search query to Bing and get back a list of suggestions. This section provides technical details about the query parameters and headers that you use to request suggestions and the JSON response objects that contain them.",
+        "operationId": "AutoSuggest",
```
Comment (jianghaolu) [inline specification/cognitiveservices/data-plane/AutoSuggest/stable/v1.0/AutoSuggest.json]: You need to reference your example in your swagger using `x-ms-examples`. You can follow the tutorial here: https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/x-ms-examples.md

**Review by jianghaolu — CHANGES_REQUESTED** [review]

File: specification/cognitiveservices/data-plane/AutoSuggest/stable/v1.0/AutoSuggest.json (hunk 1 above)
Comment (gu-wei) [reply specification/cognitiveservices/data-plane/AutoSuggest/stable/v1.0/AutoSuggest.json]: thank you for reviewing. Added

File: specification/cognitiveservices/data-plane/AutoSuggest/stable/v1.0/AutoSuggest.json (hunk 2 above)
Comment (gu-wei) [reply specification/cognitiveservices/data-plane/AutoSuggest/stable/v1.0/AutoSuggest.json]: added

**Comment by gu-wei** [comment]
@sudhansu88
Add sudhansu88 in the reviewer list

File: specification/cognitiveservices/data-plane/AutoSuggest/stable/v1.0/AutoSuggest.json [hunk 3]
```diff
@@ -0,0 +1,632 @@
...
+      "allOf": [
+        {
+          "$ref": "#/definitions/Response"
+        }
+      ],
+      "type": "object",
+      "properties": {}
+    },
+    "QueryContext": {
+      "description": "Defines the query context that Bing used for the request.",
```
Comment (jianghaolu) [inline specification/cognitiveservices/data-plane/AutoSuggest/stable/v1.0/AutoSuggest.json]: According to https://github.com/Azure/azure-rest-api-specs/pull/3180/files#diff-670740650e285f2c72a7d3ced0e1e73cR13, this either needs to have `_type` property or extend from `ResponseBase`.

File: specification/cognitiveservices/data-plane/AutoSuggest/stable/v1.0/AutoSuggest.json [hunk 4]
```diff
@@ -0,0 +1,632 @@
...
+      ],
+      "type": "object",
+      "properties": {
+        "queryContext": {
+          "$ref": "#/definitions/QueryContext",
+          "readOnly": true
+        }
+      }
+    },
+    "SuggestionsSuggestionGroup": {
```
Comment (jianghaolu) [inline specification/cognitiveservices/data-plane/AutoSuggest/stable/v1.0/AutoSuggest.json]: According to https://github.com/Azure/azure-rest-api-specs/pull/3180/files#diff-670740650e285f2c72a7d3ced0e1e73cR18, this either needs to have `_type` property or extend from `ResponseBase`.

**Review by jianghaolu — CHANGES_REQUESTED** [review]

File: specification/cognitiveservices/data-plane/AutoSuggest/stable/v1.0/AutoSuggest.json (hunk 3 above)
Comment (gu-wei) [reply specification/cognitiveservices/data-plane/AutoSuggest/stable/v1.0/AutoSuggest.json]: updated spec

File: specification/cognitiveservices/data-plane/AutoSuggest/stable/v1.0/AutoSuggest.json (hunk 4 above)
Comment (gu-wei) [reply specification/cognitiveservices/data-plane/AutoSuggest/stable/v1.0/AutoSuggest.json]: update spec, thank you

**Review by jianghaolu — APPROVED** [review]

**Pull request merged**